alerting/src/main/kotlin/org/opensearch/alerting/DocumentReturningMonitorRunner.kt

```diff
@@ -248,11 +338,22 @@ object DocumentReturningMonitorRunner : MonitorRunner {
                 val maxSeqNo: Long = docExecutionCtx.updatedLastRunContext[shard].toString().toLong()
                 logger.info("MaxSeqNo of shard_$shard is $maxSeqNo")
 
+                // If dryrun, set the previous sequence number as 1 less than the max sequence number or 0
+                val prevSeqNo = if (dryrun || monitor.id == Monitor.NO_ID)
+                    max(-1, maxSeqNo - 1)
+                else docExecutionCtx.lastRunContext[shard].toString().toLongOrNull()
+
+                if (dryrun) {
+                    logger.info("it is a dryrun")
+                }
+
+                logger.info("prevSeq: $prevSeqNo, maxSeq: $maxSeqNo")
```

Comment:
I think this has a logical bug. Since we are executing the ``runForEachQuery`` for each query in the monitor, we would like to pre-fetch the ``SeqNo`` from the ``lastRunContext`` once and use the same for every query.